estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,63 @@
+from odoo import models, fields, api, exceptions
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "property offer"
+
+    price = fields.Float()
+    status = fields.Selection(
+        copy=False,
+        selection=[
+            ("accepted", "Accepted"),
+            ("refused", "Refused"),
+        ],
+    )
+    partner_id = fields.Many2one("res.partner", required=True, string="Partner")
+    property_id = fields.Many2one(
+        "estate.property", required=True, string="Property", ondelete="cascade"
+    )
+    validity = fields.Integer(default=7)
+    deadline_date = fields.Date(
+        compute="_compute_deadline_date", inverse="_inverse_deadline_date"
+    )
+
+    _postive_price = models.Constraint(
+        "CHECK (price > 0)",
+        "The expected price must be strictly positive"
+    )
+
+    @api.depends("create_date", "validity")
+    def _compute_deadline_date(self):
+        for record in self:
+            create_date = record.create_date or fields.Date.today()
+            record.deadline_date = fields.Date.add(create_date, days=record.validity)
+
+    def _inverse_deadline_date(self):
+        for record in self:
+            create_date = record.create_date or fields.Date.today()
+            record.validity = (record.deadline_date - (create_date.date())).days
```

Comment:
```suggestion
            record.validity = (record.deadline_date - create_date.date()).days
```
I'm wondering if you really need to do .date() here, possibly the ORM will do the casting for you.